### ISSUE TITLE
sequence to sequence plans

### ISSUE THREAD

**Issue by pax7** [CodeReclaimers/neat-python#89]
Hello,

Are there any plans to include a sequence to sequence example? Something chatbot like?

**Comment by CodeReclaimers** [comment]
I would like to have something like that, I'm hoping I'll get some spare time to look into it soon. :)

**Comment by stark7** [comment]
Hi - has anyone had a chance to look at this?

**Comment by d0pa** [comment]
Well you need to be more specific what do you mean like seq to seq. Does it mean comparing two wave functions ? (not trying to be complicated)

**Comment by CodeReclaimers** [comment]
I don't know enough about natural language processing to take a reasonable stab at a chatbot using NEAT (other than just doing some kind of character-by-character predictor, which I've never tried in any fashion), so I was thinking about trying one of these: https://gym.openai.com/envs#algorithmic

**Comment by drallensmith** [comment]
I agree with starting with the openai tasks. For full natural language processing, I suspect that preprocessing with a POS (parts of speech) tagger and a stemmer (trying -> try, for instance) will be needed for any sane size of network.

@d0pa: Comparing wave functions (or comparing multiple, as opposed to outputting an altered version of, sequences of letters/DNA bases/protein) is an interesting problem - alignment of such and the more general problem of recognizing similar sequences is a major interest in bioinformatics and a number of other fields (including, as I mentioned on another thread, input space classification for measuring novelty).

**Comment by d0pa** [comment]
@drallensmith, I am working on such problem, but it gets complicated even to prove to my self, that my my work is useful what so ever. It's very time consuming to reproduce/understand similar experiments also there is trade-off in error rate versus computational burden, additionally understanding or setting up new real world like experiments, requires understanding of problem domain itself (hardest by far).  Instead of working with OpenAI gym I find it more useful to work with [UCR Archive](http://www.cs.ucr.edu/~eamonn/time_series_data/).

**Comment by drallensmith** [comment]
@d0pa: It would be nice on the UCR Archive if, perhaps by a link from the first author's name, they gave direct access to the paper they want everyone to read... Does the archive zip file contain a listing of what each sub-dataset is (and in particular the problem domain involved)?

**Comment by drallensmith** [comment]
@d0pa: BTW, one way that I keep myself motivated - and one reason that I've wound up doing research on computers as opposed to in a "wet lab" - is by trying to do things that are both needed for the research long-term and useful (to others) short-term. My work on neat-python is an example of this!

**Comment by d0pa** [comment]
@drallensmith Sorry is my fault, I should have provided newer link for [UCR Archive,](http://timeseriesclassification.com/index.php) I think this will answer your questions.

**Comment by drallensmith** [comment]
Thanks! Don't worry about it - they really should have put a (prominent) link to the new page on the old one...
